### PR TITLE
Do not log sensetive config values in debug logs.

### PR DESCRIPTION
**Pull request by medb**

### CODE REVIEW

File: util-hadoop/src/main/java/com/google/cloud/hadoop/util/HadoopConfigurationProperty.java [hunk 1]
```diff
@@ -84,7 +84,9 @@ public String getPassword(Configuration config) {
     } catch (IOException e) {
       throw new RuntimeException(e);
     }
-    return logProperty(lookupKey, value == null ? (String) defaultValue : String.valueOf(value));
+    return logProperty(
+        lookupKey,
+        RedactedString.create(value == null ? (String) defaultValue : String.valueOf(value)));
```
Comment (mprashanthsagar) [inline util-hadoop/src/main/java/com/google/cloud/hadoop/util/HadoopConfigurationProperty.java:89]: nit: extract to a var

File: util-hadoop/src/main/java/com/google/cloud/hadoop/util/HadoopCredentialConfiguration.java [hunk 2]
```diff
@@ -155,24 +155,19 @@
...
             .setServiceAccountEnabled(
                 ENABLE_SERVICE_ACCOUNTS_SUFFIX
                     .withPrefixes(keyPrefixes)
                     .get(config, config::getBoolean))
             .setServiceAccountPrivateKeyId(
-                RedactedString.create(
-                    SERVICE_ACCOUNT_PRIVATE_KEY_ID_SUFFIX
-                        .withPrefixes(keyPrefixes)
-                        .getPassword(config)))
+                SERVICE_ACCOUNT_PRIVATE_KEY_ID_SUFFIX.withPrefixes(keyPrefixes).getPassword(config))
```
Comment (mprashanthsagar) [inline util-hadoop/src/main/java/com/google/cloud/hadoop/util/HadoopCredentialConfiguration.java:167]: Should we use `RedactedString` here as well ?

File: util-hadoop/src/main/java/com/google/cloud/hadoop/util/HadoopCredentialConfiguration.java [hunk 3]
```diff
@@ -155,24 +155,19 @@
   public static CredentialFactory getCredentialFactory(
```
Comment (mprashanthsagar) [inline util-hadoop/src/main/java/com/google/cloud/hadoop/util/HadoopCredentialConfiguration.java:155]: For future proofing, could we add a test ?

**Review by mprashanthsagar — CHANGES_REQUESTED** [review]

File: util-hadoop/src/main/java/com/google/cloud/hadoop/util/HadoopCredentialConfiguration.java (hunk 2 above)
Comment (medb) [reply util-hadoop/src/main/java/com/google/cloud/hadoop/util/HadoopCredentialConfiguration.java:167]: It's used automatically - `getPassword` returns `RedactedString` now.

File: util-hadoop/src/main/java/com/google/cloud/hadoop/util/HadoopCredentialConfiguration.java [hunk 4]
```diff
@@ -187,12 +182,8 @@ public static CredentialFactory getCredentialFactory(
                 TOKEN_SERVER_URL_SUFFIX.withPrefixes(keyPrefixes).get(config, config::get))
             .setProxyAddress(
                 PROXY_ADDRESS_SUFFIX.withPrefixes(keyPrefixes).get(config, config::get))
-            .setProxyUsername(
-                RedactedString.create(
-                    PROXY_USERNAME_SUFFIX.withPrefixes(keyPrefixes).getPassword(config)))
-            .setProxyPassword(
-                RedactedString.create(
-                    PROXY_PASSWORD_SUFFIX.withPrefixes(keyPrefixes).getPassword(config)))
+            .setProxyUsername(PROXY_USERNAME_SUFFIX.withPrefixes(keyPrefixes).getPassword(config))
```
Comment (mayanks) [inline util-hadoop/src/main/java/com/google/cloud/hadoop/util/HadoopCredentialConfiguration.java:185]: overall the implementation looks good. I am just wondering if we need to redact username? printing out username sometimes is helpful. Looking at the code, I do see that it was redacted earlier. Just wondering if if is really needed.

File: util-hadoop/src/main/java/com/google/cloud/hadoop/util/HadoopCredentialConfiguration.java (hunk 3 above)
Comment (medb) [reply util-hadoop/src/main/java/com/google/cloud/hadoop/util/HadoopCredentialConfiguration.java:155]: Added.

File: util-hadoop/src/main/java/com/google/cloud/hadoop/util/HadoopCredentialConfiguration.java (hunk 4 above)
Comment (medb) [reply util-hadoop/src/main/java/com/google/cloud/hadoop/util/HadoopCredentialConfiguration.java:185]: This is already redacted `getPassword` returns redacted string

**Review by mprashanthsagar — APPROVED** [review]

**Pull request merged**